diff --git a/components/app_update/esp_ota_ops.c b/components/app_update/esp_ota_ops.c
index f77d54b5..76140d6b 100644
--- a/components/app_update/esp_ota_ops.c
+++ b/components/app_update/esp_ota_ops.c
@@ -81,16 +81,15 @@ static const esp_partition_t *read_otadata(esp_ota_select_entry_t *two_otadata)
         return NULL;
     }
 
-    esp_partition_mmap_handle_t ota_data_map;
-    const void *result = NULL;
-    esp_err_t err = esp_partition_mmap(otadata_partition, 0, otadata_partition->size, ESP_PARTITION_MMAP_DATA, &result, &ota_data_map);
+    esp_err_t err = esp_partition_read(otadata_partition, 0, &two_otadata[0], sizeof(esp_ota_select_entry_t));
     if (err != ESP_OK) {
-        ESP_LOGE(TAG, "mmap otadata filed. Err=0x%8x", err);
+        ESP_LOGE(TAG, "failed to read otadata[0]. Err=0x%8x", err);
+        return NULL;
+    }
+    err = esp_partition_read(otadata_partition, otadata_partition->erase_size, &two_otadata[1], sizeof(esp_ota_select_entry_t));
+    if (err != ESP_OK) {
+        ESP_LOGE(TAG, "failed to read otadata[1]. Err=0x%8x", err);
         return NULL;
-    } else {
-        memcpy(&two_otadata[0], result, sizeof(esp_ota_select_entry_t));
-        memcpy(&two_otadata[1], result + otadata_partition->erase_size, sizeof(esp_ota_select_entry_t));
-        esp_partition_munmap(ota_data_map);
     }
     return otadata_partition;
 }
